openedx/openedx-platform
# [ENT-370] Update Support Link on Sign In Page

As said in the relevant ticket, the Learner and Enterprise teams evaluated the current workings of the password reset page, and it was decided that it'd be more helpful if the current support link could be updated to something more specific.

**JIRA tickets**: Implements [ENT-370](https://openedx.atlassian.net/browse/ENT-370).

**Discussions**: The main change is the ability to configure the particular password reset support link to something specific to the organization's documentation. Currently, the acceptance criteria for the PR is to change the link to

- https://support.edx.org/hc/en-us/articles/206212088-What-if-I-did-not-receive-a-password-reset-message-

However, as the discussion goes, it'd be maybe better to keep it defaulting to `SUPPORT_SITE_LINK` while allowing a configuration option.

**Dependency**: There is a corresponding [configuration PR](https://github.com/edx/configuration/pull/3888) which will be needed for new setups. edx-internal will also need to update the link to point to the one above.

**Deployment targets**: edx.org and edge.edx.org

**Merge deadline**: Not that urgent.

**Sandbox**: https://pr15196.sandbox.opencraft.hosting/

**Testing instructions**:

1. Spin up a standard devstack.
2. Edit the `lms.env.json` file at `/edx/app/edxapp/lms.env.json` and configure the `PASSWORD_RESET_SUPPORT_LINK` to have your custom password reset support URL.
3. Start up LMS.
4. Register for an account. Note the email.
5. Logout.
6. Try to log back in, but press "Forgot password?" instead.
7. Put in the email of the account you created earlier.
8. When the green background notification pops up, click the "contact technical support" -- it should bring you to the link you put for `PASSWORD_RESET_SUPPORT_LINK`.
9. Log back in (no need to follow up with the reset, unless you really forgot!).
10. Go to your account settings page (i.e. https://devstack.com/account/settings).
11. Scroll down to "Reset Your Password" and press the button.
12. See the modified help text, and click "support" -- it should bring you to the link you put for `PASSWORD_RESET_SUPPORT_LINK`.

**Author notes and concerns**:

1. If one tries to reset the password through the account settings page (i.e. when already logged in), the modified help text that appears only appears for a few seconds. The user may wait a while before realizing the email isn't arriving successfully, and only then think to use that support link. But now that the link display is gone, the user will have to again press "Reset Your Password" -- probably not an intentional experience. Either we can scratch the idea to use it there as well, or not have that particular modified section revert itself.

**Reviewers**

- - -
**Settings**
```yaml
configuration_version: ent-370
edx_ansible_source_repo: https://github.com/UmanShahzad/configuration.git
EDXAPP_PASSWORD_RESET_SUPPORT_LINK: https://support.edx.org/hc/en-us/articles/206212088-What-if-I-did-not-receive-a-password-reset-message-
SWAPFILE_SIZE: 4GB
```
